### PR TITLE
get supported agents versions api

### PR DESCRIPTION
**Pull request by sigalmaya**
- Adding API that returns the supported agents' versions by the service manager. 
- The supported versions API reads the configurations agent settings.  The configuration could be injected as env variables.

### CODE REVIEW

File: api/notifications/notifications.go [hunk 1]
```diff
@@ -32,10 +32,20 @@ func (c *Controller) handleWS(req *web.Request) (*web.Response, error) {
 		return nil, errors.New("user details not found in request context")
 	}
 
+
 	platform, err := extractPlatformFromContext(user)
 	if err != nil {
 		return nil, err
 	}
+	version:= req.Header.Get("X-Version")
```
Comment (eyalnathan) [inline api/notifications/notifications.go:40]: can we have a more specific header here? e.g. `X-Peripli-Agent-Version`

File: pkg/types/platform.go [hunk 2]
```diff
@@ -53,6 +53,7 @@ type Platform struct {
 	Description       string       `json:"description"`
 	Credentials       *Credentials `json:"credentials,omitempty"`
 	OldCredentials    *Credentials `json:"old_credentials,omitempty"`
+	Version           string       `json:"-"`
```
Comment (eyalnathan) [inline pkg/types/platform.go:56]: don't we want it returned in the platforms API? same question for "active" (i know it was there before)

File: storage/postgres/platform.go [hunk 3]
```diff
@@ -42,6 +42,7 @@ type Platform struct {
 	CredentialsActive bool           `db:"credentials_active"`
 	LastActive        time.Time      `db:"last_active"`
 	Technical         bool           `db:"technical"`
+	Version           string 	     `db:"version"`
```
Comment (eyalnathan) [inline storage/postgres/platform.go:45]: indentation (or does it only look like this in the browser?)

Comment (eyalnathan) [reply storage/postgres/platform.go:45]: there a few more indentation "issues", can you `go fmt` everything?

Comment (sigalmaya) [reply storage/postgres/platform.go:45]: only in the browser

File: pkg/types/platform.go (hunk 2 above)
Comment (sigalmaya) [reply pkg/types/platform.go:56]: Are active and version field allowed to be changed via API ?

File: api/notifications/notifications.go [hunk 4]
```diff
@@ -36,6 +37,15 @@ func (c *Controller) handleWS(req *web.Request) (*web.Response, error) {
 	if err != nil {
 		return nil, err
 	}
+	version := req.Header.Get(AgentVersionHeader)
+	if version != "" {
+		platform.Version = version
+		_, err = c.repository.Update(ctx, platform, nil)
```
Comment (eyalnathan) [inline api/notifications/notifications.go:43]:
```suggestion
		if platform.Version != version {
		  platform.Version = version
		  _, err = c.repository.Update(ctx, platform, nil)
		}
```

File: api/notifications/notifications.go [hunk 5]
```diff
@@ -36,6 +37,15 @@ func (c *Controller) handleWS(req *web.Request) (*web.Response, error) {
...
+		if err != nil {
+			logger.Errorf("Could not update platform version for platform %s: %v", platform.ID, err)
+			return nil, err
+		}
```
Comment (eyalnathan) [inline api/notifications/notifications.go:47]:
```suggestion
	if platform.Version != version {	
		platform.Version = version
		_, err = c.repository.Update(ctx, platform, nil)
		if err != nil {
			logger.Errorf("Could not update platform version for platform %s: %v", platform.ID, err)
			return nil, err
		}
	}
```

File: pkg/env/env.go [hunk 6]
```diff
@@ -133,10 +134,17 @@ func (v *ViperEnv) Unmarshal(value interface{}) error {
 			return err
 		}
 	}
-
 	return v.Viper.Unmarshal(value)
+
 }
 
+func (v *ViperEnv) parseJson() {
+	versions := v.Viper.GetStringMapStringSlice("agents.versions")
```
Comment (eyalnathan) [inline pkg/env/env.go:142]: is this the right way to do it? seems weird to have code for a specific config property in here.

File: pkg/agents/agents_version.go [hunk 7]
```diff
@@ -0,0 +1,12 @@
+package agents
+
+type Settings struct {
+	SupportedVersions map[string][]string `mapstructure:"supported_versions"`
+	VersionsEnv       string              `mapstructure:"versions"`
```
Comment (eyalnathan) [inline pkg/agents/agents_version.go:5]: do we really need both?

**Review by eyalnathan — APPROVED** [review]

**Pull request merged**